apps/barista/src/components/color-component/color.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Input } from '@angular/core';
+import { DtColors } from '@dynatrace/barista-components/theming';
+import { BaCopyToClipboardService } from '../../shared/copy-to-clipboard.service';
+import { timer } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'ba-color',
+  templateUrl: 'color.html',
+  styleUrls: ['color.scss'],
+})
+export class BaColor {
+  @Input()
+  get color(): string {
+    return this._colorname;
+  }
+  set color(value: string) {
+    this._colorname = value.toUpperCase().replace('-', '_');
```

Comment:
do not modify the value you are returning in the getter. When you set a value and then get it, it has to be the same value.

In your case:
```
colorComponent.color = 'Some-thing';
console.log(colorComponent.color); // SOME-THING
```

Better (getter returns the same value i have set)

```
colorComponent.color = 'Some-thing';
console.log(colorComponent.color); // Some-thing
```
You need to either store the colorname and the transformed color name in different properties or transform it back in the getter (not ideal).